src/algebra/direct_sum_graded.lean

```diff
@@ -188,9 +194,11 @@ gcomm_monoid.of_add_submonoids (λ i, (carriers i).to_add_submonoid) one_mem mul
 
 /-! #### From `submodules`s -/
 
+variables {A : Type*}
+
 /-- Build a `ghas_one` instance for a collection of `submodule`s. -/
 @[simps one]
-def ghas_one.of_submodules {R A : Type*}
```

Comment:
This was causing lean to infer `max ? ?` as the universe of `A`, which made the new `instance`s not work.